components/channel_view/index.js

```diff
@@ -29,11 +31,16 @@ function mapStateToProps(state) {
     const config = getConfig(state);
     const enableTutorial = config.EnableTutorial === 'true';
     const tutorialStep = getInt(state, Preferences.TUTORIAL_STEP, getCurrentUserId(state), TutorialSteps.FINISHED);
+    const channel = getCurrentChannel(state);
+    const isReadOnly = channel && channel.name === Constants.DEFAULT_CHANNEL &&
+        !UserStore.isSystemAdminForCurrentUser() &&
```

Comment:
We shouldn't be using the `UserStore` directly here. You could instead import the `isCurrentUserSystemAdmin` selector from 'mattermost-redux/selectors/entities/users` and use that instead

Reply:
Also, should this be system admin or can a team admin still modify the town square?